Objects/moduleobject.c

```diff
@@ -510,10 +510,12 @@ const char *
 PyModule_GetName(PyObject *m)
 {
     PyObject *name = PyModule_GetNameObject(m);
-    if (name == NULL)
+    if (name == NULL) {
         return NULL;
+    }
+    const char *utf8 = PyUnicode_AsUTF8(name);
     Py_DECREF(name);   /* module dict has still a reference */
-    return PyUnicode_AsUTF8(name);
+    return utf8;
```

Comment:
If you consider that Py_DECREF() can destroy the name, utf8 becomes a dangling pointer after the Py_DECREF().

I don't understand the rationale of this change.

Reply:
@vstinner 

`````
const char *
PyModule_GetFilename(PyObject *m)
{
    PyObject *fileobj;
    const char *utf8;
    fileobj = PyModule_GetFilenameObject(m);
    if (fileobj == NULL)
        return NULL;
    utf8 = PyUnicode_AsUTF8(fileobj);
    Py_DECREF(fileobj);   /* module dict has still a reference */
    return utf8;
}
`````

Ah, I thought that the current sort of order differs from the conventional way.
We often Py_DECREF(obj) after some operation, so I update to follow the conventional sort of order.